File: packages/java/anip-core/src/main/java/dev/anip/core/InvokeResponse.java
package dev.anip.core;

/**
 * The server's invocation response.
 */
public class InvokeResponse {

    private final boolean success;
    private final String invocationId;
    private final String clientReferenceId;
    private final Object result;
    private final CostActual costActual;
    private final ANIPError failure;
    private final BudgetContext budgetContext;

    public InvokeResponse(boolean success, String invocationId, String clientReferenceId,
                          Object result, CostActual costActual, ANIPError failure) {
        this(success, invocationId, clientReferenceId, result, costActual, failure, null);
    }

    public InvokeResponse(boolean success, String invocationId, String clientReferenceId,
                          Object result, CostActual costActual, ANIPError failure,
                          BudgetContext budgetContext) {
        this.success = success;
        this.invocationId = invocationId;
        this.clientReferenceId = clientReferenceId;
        this.result = result;
        this.costActual = costActual;
        this.failure = failure;
        this.budgetContext = budgetContext;
    }

    public boolean isSuccess() {
        return success;
    }

    public String getInvocationId() {
        return invocationId;
    }

    public String getClientReferenceId() {
        return clientReferenceId;
    }

    public Object getResult() {
        return result;
    }

    public CostActual getCostActual() {
        return costActual;
    }

    public ANIPError getFailure() {
        return failure;
    }

    public BudgetContext getBudgetContext() {
        return budgetContext;
    }
}

File: packages/csharp/src/Anip.Core/InvokeResponse.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class InvokeResponse
{
    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("invocation_id")]
    public string InvocationId { get; set; } = "";

    [JsonPropertyName("client_reference_id")]
    public string? ClientReferenceId { get; set; }

    [JsonPropertyName("task_id")]
    public string? TaskId { get; set; }

    [JsonPropertyName("parent_invocation_id")]
    public string? ParentInvocationId { get; set; }

    [JsonPropertyName("upstream_service")]
    public string? UpstreamService { get; set; }

    [JsonPropertyName("result")]
    public object? Result { get; set; }

    [JsonPropertyName("cost_actual")]
    public CostActual? CostActual { get; set; }

    [JsonPropertyName("failure")]
    public AnipError? Failure { get; set; }

    [JsonPropertyName("budget_context")]
    public BudgetContext? BudgetContext { get; set; }
}
